shared/events.js

```diff
@@ -348,14 +392,23 @@ export class PostMessage extends HashableAction {
   static fields = PostMessage.Fields([
     ['from', 'string'],
     ['headers', 'string', 'repeated'],
-    ['message', 'string']
+    ['message', 'string'],
+    ['sentDate', 'string'],
+    ['read', 'bool']
   ])
+  constructor (data: JSONObject, parentHash?: string) {
+    super({...data, read: false}, parentHash)
+  }
 }
-
 export class PostInvite extends HashableAction {
   static fields = PostInvite.Fields([
-    ['groupId', 'string']
+    ['groupId', 'string'],
+    ['sentDate', 'string'],
+    ['read', 'bool']
   ])
+  constructor (data: JSONObject, parentHash?: string) {
+    super({...data, read: false}, parentHash)
+  }
```

Comment:
`PostInvite` looks totally unnecessary to me, and violation of DRY, it's basically the same as `PostMessage`.

No need to create it. Just use `PostMessage`, and make the `message` field JSON that contains your `groupId`. Every message can have a `type` field in the JSON, which can specify what sort of message it is, and one of those can be a `"groupInvite"`.